apps/wallet-mobile/src/features/Receive/common/SmallAddressCard/SmallAddressCard.tsx

```diff
@@ -0,0 +1,164 @@
+import {useTheme} from '@yoroi/theme'
+import React from 'react'
+import {StyleSheet, Text, TouchableOpacity, View} from 'react-native'
+import LinearGradient from 'react-native-linear-gradient'
+import Animated, {FadeInDown, FadeInUp, FadeOut, FadeOutDown, Layout} from 'react-native-reanimated'
+
+import {useCopy} from '../../../../../src/legacy/useCopy'
+import {SkeletonSmallCardAddress} from '../SkeletonAddressDetail/SkeletonAddressDetail'
+import {useStrings} from '../useStrings'
+
+export type SmallAddressCardProps = {
+  address: string
+  isUsed: boolean
+  loading: boolean
+  date?: string
+  onPress?: () => void
+}
+
+export function SmallAddressCard({address, isUsed, date, onPress, loading}: SmallAddressCardProps) {
+  const strings = useStrings()
+  const [isCopying, copy] = useCopy()
+
+  const {styles, colors} = useStyles()
+
+  return (
+    <>
+      {loading ? (
+        <View style={styles.skeleton}>
+          <SkeletonSmallCardAddress />
+        </View>
+      ) : (
+        <Animated.View layout={Layout} entering={FadeInUp} exiting={FadeOut}>
+          <TouchableOpacity
+            style={styles.smallAddressCard}
+            activeOpacity={0.6}
+            onLongPress={() => copy(address)}
+            onPress={onPress}
+          >
+            <LinearGradient
+              style={[StyleSheet.absoluteFill, {opacity: 1}]}
+              start={{x: 0, y: 0}}
+              end={{x: 0, y: 1}}
+              colors={colors.bgCard}
+            />
+
+            <Text style={styles.textAddress}>{address}</Text>
+
+            <View style={styles.footer}>
+              <View style={isUsed ? styles.statusUsed : styles.statusUnused}>
+                <Text style={isUsed ? styles.statusUsedText : styles.statusUnusedText}>
+                  {isUsed ? strings.usedAddress : strings.unusedAddress}
+                </Text>
+              </View>
+
+              {isUsed && <Text style={styles.date}>{date}</Text>}
+            </View>
+          </TouchableOpacity>
+        </Animated.View>
+      )}
+
+      {isCopying && (
+        <Animated.View layout={Layout} entering={FadeInDown} exiting={FadeOutDown} style={styles.isCopying}>
+          <Text style={styles.textCopy}>{strings.addressCopiedMsg}</Text>
+        </Animated.View>
+      )}
+    </>
+  )
+}
+
+const useStyles = () => {
+  const {theme} = useTheme()
+
+  const styles = StyleSheet.create({
+    smallAddressCard: {
+      borderRadius: 8,
+      width: '100%',
```

Comment:
avoid %

Reply:
I need to use % on it

Reply:
it can be achieved if wrapped on flex grow 1 view.